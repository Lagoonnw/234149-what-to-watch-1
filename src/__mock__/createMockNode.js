export function createNodeMock(element) {
  if (element.type === `video`) {
    return {};
  }
  return null;
}
