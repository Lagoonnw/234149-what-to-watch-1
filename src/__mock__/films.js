export const films = [
  {
    id: 54585444,
    name: `Shutter Island`,
    genre: `thriller`,
    poster: `img/shutter-island.jpg`,
    src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    id: 4587653,
    name: `Pulp Fiction`,
    genre: `crime`,
    poster: `img/pulp-fiction.jpg`,
    src: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },
  {
    id: 54546847,
    name: `Snatch`,
    genre: `crime`,
    poster: `img/snatch.jpg`,
    src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    id: 54545454545,
    name: `Bohemian Rhapsody`,
    genre: `drama`,
    poster: `img/bohemian-rhapsody.jpg`,
    src: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },
  {
    id: 549963331,
    name: `Aviator`,
    genre: `family`,
    poster: `img/aviator.jpg`,
    src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  }
];

