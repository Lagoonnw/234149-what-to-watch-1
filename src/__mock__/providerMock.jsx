import React from 'react';
import {createStore} from 'redux';
import {Provider} from 'react-redux';
import {genresList} from '../constants/constants';
import PropTypes from 'prop-types';

const mock = [
  {
    id: 54545454,
    name: `Snatch`,
    genre: `horror`,
    poster: `pic.jpg`,
    src: `video.mp4`,
    onMouseLeave: jest.fn(),
    onMouseEnter: jest.fn()
  },
  {
    id: 9898988,
    name: `Pulp Fiction`,
    genre: `horror`,
    poster: `pic.jpg`,
    src: `video.mp4`,
    onMouseLeave: jest.fn(),
    onMouseEnter: jest.fn()
  },
  {
    id: 545468787,
    name: `The Witcher`,
    genre: `horror`,
    poster: `pic.jpg`,
    src: `video.mp4`,
    onMouseLeave: jest.fn(),
    onMouseEnter: jest.fn()
  }
];
const initialStateMock = {
  activeGenre: null,
  movies: mock,
  sortedMovies: mock,
  genres: Array.from(genresList)
};
const reducer = (state = initialStateMock) => state;
const store = createStore(reducer);

export const ProviderMock = ({children}) => <Provider store={store}>{children}</Provider>;

ProviderMock.propTypes = {
  children: PropTypes.object
};
