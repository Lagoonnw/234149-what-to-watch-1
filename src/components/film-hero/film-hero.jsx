import React from 'react';
import PropTypes from 'prop-types';
import Header from '../header/header.jsx';
import {Link} from 'react-router-dom';

export const FilmHero = ({backgroundImage, name, genre, released, id, onPlayClick, saveMovieToFavorite, history}) => {
  const onClick = () => onPlayClick(id);
  const onAddBtnClick = () => {
    saveMovieToFavorite(id);
    history.push(`/myList`);
  };


  return (
    <div className="movie-card__hero" >
      <div className="movie-card__bg">
        <img src={backgroundImage} alt="The Grand Budapest Hotel"/>
      </div>

      <h1 className="visually-hidden">WTW</h1>

      <Header />
      <div className="movie-card__wrap" >
        <div className="movie-card__desc">
          <h2 className="movie-card__title">{name}</h2>
          <p className="movie-card__meta">
            <span className="movie-card__genre">{genre}</span>
            <span className="movie-card__year">{released}</span>
          </p>

          <div className="movie-card__buttons">
            <button className="btn btn--play movie-card__button" type="button" onClick={onClick}>
              <svg viewBox="0 0 19 19" width="19" height="19">
                <use xlinkHref="#play-s"></use>
              </svg>
              <span>Play</span>
            </button>
            <button className="btn btn--list movie-card__button" type="button" onClick={onAddBtnClick}>
              <svg viewBox="0 0 19 20" width="19" height="20">
                <use xlinkHref="#add"></use>
              </svg>
              <span>My list</span>
            </button>
            <Link to={`/film/${id}/review`} className="btn movie-card__button">Add review</Link>
          </div>
        </div>
      </div>
    </div>
  );
};

FilmHero.propTypes = {
  id: PropTypes.number.isRequired,
  backgroundImage: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  genre: PropTypes.string.isRequired,
  released: PropTypes.number.isRequired,
  onPlayClick: PropTypes.func.isRequired,
  saveMovieToFavorite: PropTypes.func.isRequired,
  history: PropTypes.object.isRequired
};
