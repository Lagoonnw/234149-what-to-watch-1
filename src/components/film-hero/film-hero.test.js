import React from 'react';
import renderer from 'react-test-renderer';
import {FilmHero} from './film-hero';
import {ProviderMock} from '../../__mock__/providerMock';

const mockProps = {
  backgroundImage: `#F5F5F5`,
  name: `Movie`,
  genre: `Action`,
  released: 1992
};

test(`Should render Film hero correctly`, () => {
  const Component = () => (
    <ProviderMock><FilmHero {...mockProps}/></ProviderMock>
  );
  const tree = renderer.create(<Component/>).toJSON();
  expect(tree).toMatchSnapshot();
});
