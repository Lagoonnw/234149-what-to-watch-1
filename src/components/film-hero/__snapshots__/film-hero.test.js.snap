// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Film hero correctly 1`] = `
<div
  className="movie-card__hero"
>
  <div
    className="movie-card__bg"
  >
    <img
      alt="The Grand Budapest Hotel"
      src="#F5F5F5"
    />
  </div>
  <h1
    className="visually-hidden"
  >
    WTW
  </h1>
  <header
    className="page-header undefined"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="user-block"
    >
      <a
        className="user-block__link"
        href="/login"
        onClick={[Function]}
      >
        Sign in
      </a>
    </div>
  </header>
  <div
    className="movie-card__wrap"
  >
    <div
      className="movie-card__desc"
    >
      <h2
        className="movie-card__title"
      >
        Movie
      </h2>
      <p
        className="movie-card__meta"
      >
        <span
          className="movie-card__genre"
        >
          Action
        </span>
        <span
          className="movie-card__year"
        >
          1992
        </span>
      </p>
      <div
        className="movie-card__buttons"
      >
        <button
          className="btn btn--play movie-card__button"
          onClick={[Function]}
          type="button"
        >
          <svg
            height="19"
            viewBox="0 0 19 19"
            width="19"
          >
            <use
              xlinkHref="#play-s"
            />
          </svg>
          <span>
            Play
          </span>
        </button>
        <button
          className="btn btn--list movie-card__button"
          type="button"
        >
          <svg
            height="20"
            viewBox="0 0 19 20"
            width="19"
          >
            <use
              xlinkHref="#add"
            />
          </svg>
          <span>
            My list
          </span>
        </button>
        <a
          className="btn movie-card__button"
          href="add-review.html"
        >
          Add review
        </a>
      </div>
    </div>
  </div>
</div>
`;
